x-pack/legacy/plugins/maps/public/layers/styles/vector/components/vector_style_editor.js

```diff
@@ -302,4 +306,25 @@ export class VectorStyleEditor extends Component {
       </Fragment>
     );
   }
+
+  render() {
+    return (
+      <Fragment>
+        {this._renderProperties()}
+
```

Comment:
In hindsight, maybe we should reconsider and omitting this from this first version.

 The UX is getting more complex, and this is a very subtle option to how the layer and styling functions in context of dates. And for example how it interacts with the dynamic scaling.

- the switch is only relevant _if_ date-fields are driving the style (so should be hidden/greyed out if that's not the case).
- this is global here, but the sigma-configuration are local to the style-property.
- index patterns without date-field should not have it
- should index-patterns _with_ date-fields but _without_ a default date-field even have this? (because right now, this is how we "link" layers conceptually to the global time-picker, ie. by the presence of a default-time-field. Now we introduce another way). (in that scenario right now, this switch is a no-op).

I feel like it introduces quite some complexity that we might want to avoid, at least right now introducing this feature.

Reply:
vector style isTimeAware effects all fields, not just date fields. This value determines whether the field meta request is filtered by the global time range or not. I will update the UI to only show the option when the layer contains dynamic styles with field meta enabled.